main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -54,8 +116,34 @@ object CompleteProvider {
+    CompletionItem("enum", "enum ${1:name} {\n    case ${2:DataType1}\n    ${4:/* add more algebric data types separated by comma (,) */}\n}", None, Some("code snippet for enum"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("if", "if (${1:/* condition */}) {\n    ${2:/* code */}\n}", None, Some("code snippet for if ()"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import ${1:method}", None, Some("import code snippet for method call"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import new ${1:object} as ${2:name}", None, Some("import code snippet for new object"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import get ${1:field} as ${2:name}", None, Some("import code snippet to get field"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import set ${1:field} as ${2:name}", None, Some("import code snippet to set field"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("instance", "instance ${1:name}[Option[${2:a}]] with $1[$2] {\n    ${3:/* code */}\n}", None, Some("code snippet for instance"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("let", "let ${1:name} = ${2:exp};\n", None, Some("code snippet for let"), CompletionItemKind.Snippet, InsertTextFormat.PlainText, Nil),
+    CompletionItem("let*", "let* ${1:name} = ${2:exp};\n", None, Some("code snippet for let*"), CompletionItemKind.Snippet, InsertTextFormat.PlainText, Nil),
+    CompletionItem("main", "def main(args: Array[String]) : Int32 & Impure = \n    ${1:/* code */} \n    0", None, Some("code snippet for main ()"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("main Hello World", "def main(_args: Array[String]) : Int32 & Impure = \n    println(\"Hello World!\");\n    0", None, Some("code snippet for Hello World Program"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("match", "match ${1:exp} {\n    case ${2:pat} => ${3:exp}\n}", None, Some("code snippet for pattern match"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("namespace", "namespace ${1:name} {\n    ${2:/* code */} \n}", None, Some("code snippet to create namespace"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("opaque", "opaque type ${1:name} = ${2:type}", None, Some("code snippet for opaque type"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("pub def", "pub def ${1:name}(${2:arg}:${3:type}): ${4:return_type} = \n", None, Some("code snippet to define a public function"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
```

Comment:
Several things can be pub, so we need to figure out how to deal with that.

Reply:
okay. removing it for now.